### PR TITLE
fix: updated msrv and fix clippy lints

### PR DESCRIPTION
**Pull request by aawsome**
`ahash` bumped their MSRV, we also bump ours now so other PRs are unblocked from CI failing.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ members = ["crates/core", "crates/backend", "crates/testing", "examples/*"]
 resolver = "2"
 
 [workspace.package]
-rust-version = "1.71.1"
+rust-version = "1.72.0"
```
Comment (simonsan) [inline Cargo.toml:6]: Not a problem, but interested what happened?

Comment (aawsome) [reply Cargo.toml:6]: see https://github.com/rustic-rs/rustic_core/actions/runs/7869482808/job/21468685181?pr=154

Comment (simonsan) [reply Cargo.toml:6]: Interesting!

```console
ahash v0.8.7
├── ahash feature "default"
│   └── quick_cache v0.4.1
│       ├── quick_cache feature "ahash"
│       │   └── quick_cache feature "default"
│       │       └── rustic_core v0.2.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\crates\core)
│       │           └── rustic_core feature "default" (command-line)
│       │               ├── backup v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\backup)
│       │               │   └── backup feature "default" (command-line)
│       │               ├── check v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\check)
│       │               │   └── check feature "default" (command-line)
│       │               ├── config v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\config)
│       │               │   └── config feature "default" (command-line)
│       │               ├── copy v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\copy)
│       │               │   └── copy feature "default" (command-line)
│       │               ├── forget v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\forget)
│       │               │   └── forget feature "default" (command-line)
│       │               ├── init v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\init)
│       │               │   └── init feature "default" (command-line)
│       │               ├── key v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\key)
│       │               │   └── key feature "default" (command-line)
│       │               ├── ls v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\ls)
│       │               │   └── ls feature "default" (command-line)
│       │               ├── merge v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\merge)
│       │               │   └── merge feature "default" (command-line)
│       │               ├── prune v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\prune)
│       │               │   └── prune feature "default" (command-line)
│       │               ├── restore v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\restore)
│       │               │   └── restore feature "default" (command-line)
│       │               ├── rustic_backend v0.1.1 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\crates\backend)
│       │               │   ├── rustic_backend feature "default" (command-line)
│       │               │   │   ├── backup v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\backup) (*)
│       │               │   │   ├── check v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\check) (*)
│       │               │   │   ├── config v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\config) (*)
│       │               │   │   ├── copy v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\copy) (*)
│       │               │   │   ├── forget v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\forget) (*)
│       │               │   │   ├── init v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\init) (*)
│       │               │   │   ├── key v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\key) (*)
│       │               │   │   ├── ls v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\ls) (*)
│       │               │   │   ├── merge v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\merge) (*)
│       │               │   │   ├── prune v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\prune) (*)
│       │               │   │   ├── restore v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\restore) (*)
│       │               │   │   └── tag v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\tag)
│       │               │   │       └── tag feature "default" (command-line)
│       │               │   │   [dev-dependencies]
│       │               │   │   └── rustic_core v0.2.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\crates\core) (*)
│       │               │   ├── rustic_backend feature "opendal"
│       │               │   │   ├── rustic_backend feature "default" (command-line) (*)
│       │               │   │   ├── rustic_backend feature "s3"
│       │               │   │   │   └── rustic_backend feature "default" (command-line) (*)
│       │               │   │   └── rustic_backend feature "sftp"
│       │               │   │       └── rustic_backend feature "default" (command-line) (*)
│       │               │   ├── rustic_backend feature "rclone"
│       │               │   │   └── rustic_backend feature "default" (command-line) (*)
│       │               │   ├── rustic_backend feature "rest"
│       │               │   │   ├── rustic_backend feature "default" (command-line) (*)
│       │               │   │   └── rustic_backend feature "rclone" (*)
│       │               │   ├── rustic_backend feature "s3" (*)
│       │               │   ├── rustic_backend feature "sftp" (*)
│       │               │   └── rustic_backend feature "tokio"
│       │               │       └── rustic_backend feature "opendal" (*)
│       │               └── tag v0.1.0 (C:\Users\dailyuse\backup\PARA\Personal\2 Areas\Developer\Maintenance\rustic\core\examples\tag) (*)
│       ├── quick_cache feature "default" (*)
│       └── quick_cache feature "parking_lot"
│           └── quick_cache feature "default" (*)
├── ahash feature "getrandom"
│   └── ahash feature "runtime-rng"
│       └── ahash feature "default" (*)
├── ahash feature "runtime-rng" (*)
└── ahash feature "std"
    └── ahash feature "default" (*)
```

So we pull in `ahash` via `quick_cache`'s default features:

> Hasher
> By default the crate uses ahash, which is enabled (by default) via a crate feature with the same name. If the ahash feature is disabled the crate defaults to the std lib implementation instead (currently Siphash13). Note that a custom hasher can also be provided if desirable.

src.: https://docs.rs/quick_cache/0.4.1/quick_cache/#hasher

Is that a sensitive in the way that we might break something with a hash-collision here and which would value switching to the default `Siphash13`? (also with regards to discussion in #154 )

Comment (aawsome) [reply Cargo.toml:6]: Hash collision in a hashmap would only degenerate the hashmap spoiling its performance assuptions. And ahash is used as it performs generally better, AFAIR.

Comment (aawsome) [reply Cargo.toml:12]: I think we should just let it be ahash and increase the MSRV. I updated the README's

Comment (simonsan) [reply Cargo.toml:12]: I understand, also read up on it here: https://github.com/tkaitchuck/aHash/blob/master/FAQ.md#why-not-use-a-cryptographic-hash-in-a-hashmap

**Review by simonsan — APPROVED** [review]

**Pull request merged**